src/main/java/com/zombiecastlerush/util/Prompter.java

```diff
@@ -87,47 +90,82 @@ static void advanceGame(Player player) throws JsonProcessingException {
+                                    getUserInput("\nPrepare for COMBAT... press enter to continue");
+                                    combat(player, new Enemy("Zombie"), action);
+                                } else {
+                                    System.out.println("There is no Monster in the room");
+                                    break;
+                                }
+                            } else {
+                                System.out.println("There is no Monster in the room");
+                                break;
+                            }
+                            break;
+                        case "quit":
                             Game.getInstance().stop();
                             break;
-                        }
-
                     }
                     break;
             }
-        }else
+        } else {
             Game.getInstance().showInstructions();
+        }
     }
 
-
     static void solvePuzzle(Room room) {
         Puzzle puzzle = (Puzzle) room.getChallenge();
-        System.out.println(Parser.YELLOW+"Here is your puzzle....Remember you only have " + (3 - puzzle.getAttempts()) + " tries!"+Parser.ANSI_RESET);
+        System.out.println(Parser.YELLOW + "Here is your puzzle....Remember you only have " + (3 - puzzle.getAttempts()) + " tries!" + Parser.ANSI_RESET);
         puzzle.attemptPuzzle(getUserInput(puzzle.getQuestion()));
         if (puzzle.getAttempts() < 3 && !puzzle.isCleared())
             solvePuzzle(room);
         else if (puzzle.isCleared()) {
-            System.out.println(Parser.GREEN+"Right answer. You can now move to the available rooms");
+            System.out.println(Parser.GREEN + "Right answer. You can now move to the available rooms");
             if (puzzle.getInventory().getItems().size() > 0) {
-                System.out.println(puzzle.getDescription() + " drops " + puzzle.getInventory().toString() + "\n"+Parser.ANSI_RESET);
-
+                System.out.println(puzzle.getDescription() + " drops " + puzzle.getInventory().toString() + "\n" + Parser.ANSI_RESET);
                 puzzle.getInventory().transferItem(
                         puzzle.getInventory(),
                         room.getInventory(),
                         puzzle.getInventory().getItems().toArray(new Item[0])
                 );
             }
         } else {
-            System.out.println(Parser.RED+"Wrong Answer!! You have had your chances...You failed...Game Over!!!"+Parser.ANSI_RESET);
+            System.out.println(Parser.RED + "Wrong Answer!! You have had your chances...You failed...Game Over!!!" + Parser.ANSI_RESET);
             Game.getInstance().stop();
         }
     }
 
-    public static void displayCurrentScene(Player player){
+    public static void combat(Role player, Role enemy, String action) {
```

Comment:
what are you doing with the action parameter here?

Reply:
removed action argument